packages/task/src/browser/task-service.ts

```diff
@@ -409,6 +409,40 @@ export class TaskService implements TaskConfigurationClient {
     }
 
     async runTask(task: TaskConfiguration, option?: RunTaskOption): Promise<TaskInfo | undefined> {
+        const runningTasksInfo: TaskInfo[] = await this.getRunningTasks();
+
+        // check if the task is active
+        const matchedRunningTaskInfo = runningTasksInfo.find(taskInfo => {
+            const taskConfig = taskInfo.config;
+            return this.taskDefinitionRegistry.compareTasks(taskConfig, task);
+        });
+        if (matchedRunningTaskInfo) { // the task is active
+            const taskName = this.taskNameResolver.resolve(task);
+            const taskId = matchedRunningTaskInfo.taskId;
+            const terminalId = matchedRunningTaskInfo.terminalId;
+            let terminal: TerminalWidget | undefined;
+            if (terminalId) {
+                terminal = this.terminalService.getById(this.getTerminalWidgetId(terminalId));
+                if (terminal) {
+                    terminal.show(); // bring the terminal to the top
+                }
+            }
+            this.messageService.info(`The task \'${taskName}\' is already active`, 'Terminate Task', 'Restart Task').then(actionTask => {
+                if (actionTask) {
+                    this.taskServer.kill(taskId);
```

Comment:
Maybe it makes sense to use [the corresponding method](https://github.com/eclipse-theia/theia/blob/c4b4d3995ba81a80e36cd62d290dce2b23c8fd82/packages/task/src/browser/task-service.ts#L683) for terminating a task, this one handles errors